openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -693,23 +710,37 @@ impl BackingPrivate for TdxBacked {
             .map(|synic| synic.add_vp(params.vp_info.base.vp_index));
 
         Ok(Self {
-            efer: regs.efer,
-            cr0: VirtualRegister::new(ShadowedRegister::Cr0, regs.cr0, None),
-            cr4: VirtualRegister::new(ShadowedRegister::Cr4, regs.cr4, Some(allowed_cr4_bits)),
+            efer: VtlArray::new(regs.efer),
+            cr0: VtlArray::from_fn(|vtl| {
+                VirtualRegister::new(
+                    ShadowedRegister::Cr0,
+                    vtl.try_into().unwrap(),
+                    regs.cr0,
+                    None,
+                )
+            }),
+            cr4: VtlArray::from_fn(|vtl| {
+                VirtualRegister::new(
+                    ShadowedRegister::Cr4,
+                    vtl.try_into().unwrap(),
+                    regs.cr4,
+                    Some(allowed_cr4_bits),
+                )
+            }),
             direct_overlays_pfns: overlays.try_into().unwrap(),
             direct_overlay_pfns_handle: pfns_handle,
             untrusted_synic,
             eoi_exit_bitmap: [0; 4],
-            tpr_threshold: 0,
+            tpr_threshold: VtlArray::new(0),
             processor_controls,
             secondary_processor_controls,
-            interruption_information: Default::default(),
-            interruption_set: false,
-            exception_error_code: 0,
+            interruption_information: VtlArray::from_fn(|_| Default::default()),
```

Comment:
meta: seems reasonable for VtlArray to have a `Default` impl that forwards to `T::default`

Reply:
https://github.com/microsoft/openvmm/pull/589